sync.go

```diff
@@ -30,35 +58,57 @@ const (
 	ErrorCodeDeadlineExceeded
 )
 
-func (lw *LibWallet) AddSyncProgressListener(syncProgressListener SyncProgressListener) {
-	lw.syncProgressListeners = append(lw.syncProgressListeners, syncProgressListener)
+func (lw *LibWallet) initActiveSyncData() {
+	headersFetchProgress := HeadersFetchProgressReport{}
+	headersFetchProgress.GeneralSyncProgress = &GeneralSyncProgress{}
+
+	addressDiscoveryProgress := AddressDiscoveryProgressReport{}
+	addressDiscoveryProgress.GeneralSyncProgress = &GeneralSyncProgress{}
+
+	headersRescanProgress := HeadersRescanProgressReport{}
+	headersRescanProgress.GeneralSyncProgress = &GeneralSyncProgress{}
+
+	var targetTimePerBlock int32
+	if lw.activeNet.Name == "mainnet" {
+		targetTimePerBlock = MainNetTargetTimePerBlock
+	} else {
+		targetTimePerBlock = TestNetTargetTimePerBlock
+	}
+
+	activeSyncData := &activeSyncData{
+		targetTimePerBlock: targetTimePerBlock,
+
+		headersFetchProgress:     headersFetchProgress,
+		addressDiscoveryProgress: addressDiscoveryProgress,
+		headersRescanProgress:    headersRescanProgress,
+
+		beginFetchTimeStamp:     -1,
+		headersFetchTimeSpent:   -1,
+		totalDiscoveryTimeSpent: -1,
+	}
+
+	lw.syncData.activeSyncData = activeSyncData
 }
 
-func (lw *LibWallet) AddEstimatedSyncProgressListener(syncProgressListener EstimatedSyncProgressListener, logEstimatedProgress bool) {
-	syncProgressEstimator := SetupSyncProgressEstimator(
-		lw.activeNet.Params.Name,
-		logEstimatedProgress,
-		lw.GetBestBlock,
-		lw.GetBestBlockTimeStamp,
-		syncProgressListener,
-	)
+func (lw *LibWallet) AddSyncProgressListener(syncProgressListener SyncProgressListener) {
```

Comment:
Also add a `RemoveSyncProgressListener` method. I used a map (dictionary) on dcrios to enable removing listeners whenever necessary. You pass a string `identifier` when registering and deregistering a listener.